### PR TITLE
chore(all): disable promise/avoid-new eslint rule

### PR DESCRIPTION
**Pull request by szymonmaslowski**
# Checklist

- [ ] JIRA - \<link>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Disabling the `promise/avoid-new` eslint rule for the whole repo and removing all realted disable comments.

### CODE REVIEW

File: packages/core/src/ui/components/SharedWallet/AddCoSigners/AddCoSigners.stories.tsx [hunk 1]
```diff
@@ -51,10 +50,12 @@ const validateAddress: ValidateAddress = async (address) => {
...
       }
-      timeout = setTimeout(() => {
+      const twoSeconds = 2000;
+      timeout = window.setTimeout(() => {
         clearTimeout(timeout);
-        resolve(Math.random() < 0.5 ? { isValid: true, handleResolution } : { isValid: false });
-      }, 2000);
+        const factor = 0.5;
+        resolve(Math.random() < factor ? { isValid: true, handleResolution } : { isValid: false });
+      }, twoSeconds);
```
Comment (szymonmaslowski) [inline packages/core/src/ui/components/SharedWallet/AddCoSigners/AddCoSigners.stories.tsx:58]: I have made additional unrelated fixes here, as those places had errors.

**Review by danielmain — APPROVED** [review]
Great job @szymonmaslowski

**Review by pczeglik-iohk — APPROVED** [review]
Thank you @szymonmaslowski for this change. 
This is a great example of being a pro-active member of our Tribe!

File: packages/core/src/ui/components/SharedWallet/AddCoSigners/AddCoSigners.stories.tsx (hunk 1 above)
Comment (pczeglik-iohk) [reply packages/core/src/ui/components/SharedWallet/AddCoSigners/AddCoSigners.stories.tsx:58]: to me this is a transparent change. Removed magic numbers. Logic stayed the same. Right @szymonmaslowski ?

Comment (pczeglik-iohk) [reply packages/core/src/ui/components/SharedWallet/AddCoSigners/AddCoSigners.stories.tsx:58]: I'm ok with merging this without asking QA for verification if pipeline is green

Comment (szymonmaslowski) [reply packages/core/src/ui/components/SharedWallet/AddCoSigners/AddCoSigners.stories.tsx:58]: I agree with @pczeglik-iohk. Moreover this is just a storybook file.
